eventarc/generic/main_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestGenericCloudEvent(t *testing.T) {
+	tests := []struct {
+		want string
+		omit string
+	}{
+		{want: "Event received!"},
+		{want: `"Ce-Id": "1234"`},
+		{omit: `"super-secret-value"`},
+		{want: `{"message": "some string"}`},
+	}
+	log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime))
+	for _, test := range tests {
+		r, w, _ := os.Pipe()
+		log.SetOutput(w)
+		defer log.SetOutput(os.Stderr)
+
+		jsonStr := fmt.Sprintf(`{"message": "some string"}`)
+		payload := strings.NewReader(jsonStr)
+
+		req := httptest.NewRequest("POST", "/", payload)
+		req.Header.Set("Ce-Id", "1234")
+		req.Header.Set("Authorization", "super-secret-value")
+		req.Header.Set("Ce-Source", "//storage.googleapis.com/projects/YOUR-PROJECT")
```

Comment:
Consider including the header in the test case. That way, it's clear what is being tested against. On first read, it's unclear where `1234`, `super-secret-value`, etc. come from.

Reply:
I'm not sure if we're keeping this in the first place, so I think it's satisfactory for now.